eva/binder/statement_binder.py

```diff
@@ -90,31 +90,30 @@ def _bind_create_index_statement(self, node: CreateIndexStatement):
         # TODO: create index currently only works on TableInfo, but will extend later.
         assert node.table_ref.is_table_atom(), "Index can only be created on Tableinfo"
 
-        if IndexType.is_faiss_index_type(node.index_type):
-            if not node.udf_func:
-                # Feature table type needs to be float32 numpy array.
-                col_def = node.col_list[0]
-                table_ref_obj = node.table_ref.table.table_obj
-                col = [
-                    col for col in table_ref_obj.columns if col.name == col_def.name
-                ][0]
-                if not col.array_type == NdArrayType.FLOAT32:
-                    raise BinderError("Index input needs to be float32.")
-                if not len(col.array_dimensions) == 2:
-                    raise BinderError("Index input needs to be 2 dimensional.")
-            else:
-                # Output of the UDF should be 2 dimension and float32 type.
-                catalog_manager = CatalogManager()
-                udf_obj = catalog_manager.get_udf_catalog_entry_by_name(
-                    node.udf_func.name
-                )
-                for output in udf_obj.outputs:
-                    if not output.array_type == NdArrayType.FLOAT32:
-                        raise BinderError("Index input needs to be float32.")
-                    if not len(output.array_dimensions) == 2:
-                        raise BinderError("Index input needs to be 2 dimensional.")
+        assert IndexType.is_faiss_index_type(
```

Comment:
Just curious, if we support index type other than Faiss later, we still need to do it with if else right (assuming we have some index specific checks)?

Reply:
yes